packages/typespec-ts/src/transform/transformParameters.ts

```diff
@@ -256,122 +260,101 @@ function transformBodyParameters(
     dpgContext,
     bodyType,
     parameters,
     importedModels,
-    headers,
-    hasBinaryContent
+    headers
   );
 }
 
-function transformNormalBody(
+function transformRequestBody(
   dpgContext: SdkContext,
   bodyType: Type,
   parameters: HttpOperationParameters,
   importedModels: Set<string>,
-  headers: ParameterMetadata[],
-  hasBinaryContent: boolean
+  headers: ParameterMetadata[]
 ) {
-  const descriptions = extractDescriptionsFromBody(
-    dpgContext,
-    bodyType,
-    parameters
-  );
-  if (hasBinaryContent) {
-    descriptions.push("Value may contain any sequence of octets");
-  }
-  const type = extractNameFromTypeSpecType(
-    dpgContext,
-    bodyType,
+  const contentTypes = extractMediaTypes(parameters.body?.contentTypes ?? []);
+  const schema = getSchemaForType(dpgContext, bodyType, {
+    mediaTypes: contentTypes,
+    isRequestBody: true,
+    usage: [SchemaContext.Input, SchemaContext.Exception]
+  });
+
+  const descriptions = getBodyDescriptions(dpgContext, schema, parameters);
+  const type = getRequestBodyType(
+    schema,
     [SchemaContext.Input],
     importedModels,
     headers
   );
-  let schema = getSchemaForType(dpgContext, bodyType);
-  let overrideType = undefined;
-  if (hasBinaryContent) {
-    schema = enrichBinaryTypeInBody(schema);
-    overrideType = schema.typeName;
-  }
+
   return {
     isPartialBody: false,
+    hasFileTypeIncluded: isFileUploadCase(),
     body: [
       {
         properties: schema.properties,
         typeName: schema.name,
         name: "body",
-        type: overrideType ?? type,
+        type: type,
         required: parameters?.bodyParameter?.optional === false,
         description: descriptions.join("\n\n"),
         oriSchema: schema
       }
     ]
   };
-}
+  function isFileUploadCase() {
+    const isMultipartForm =
+      hasMediaType(KnownMediaType.Multipart, contentTypes) &&
+      contentTypes.length === 1;
+    let hasFileType = false;
+    if (schema.type === "object" && schema.properties) {
+      for (const p of Object.values(schema.properties)) {
+        if ((p as Schema).typeName?.includes(BINARY_AND_FILE_TYPE_UNION)) {
+          hasFileType = true;
+          break;
+        }
+      }
+    }
 
-function getBodyDetail(
-  dpgContext: SdkContext,
-  bodyType: Type,
-  headers: ParameterMetadata[]
-) {
-  const contentTypes: string[] = headers
-    .filter((h) => h.name === "contentType")
-    .map((h) => {
-      return getTypeName(h.param, [SchemaContext.Input]);
-    });
-  const hasBinaryContent = contentTypes.some((c) =>
-    isBinaryPayload(dpgContext, bodyType, c)
-  );
-  const hasFormContent = contentTypes.includes(`"multipart/form-data"`);
-  return { hasBinaryContent, hasFormContent };
+    return isMultipartForm && hasFileType;
+  }
 }
 
-function extractNameFromTypeSpecType(
-  dpgContext: SdkContext,
-  type: Type,
+function getRequestBodyType(
+  bodySchema: Schema,
   schemaUsage: SchemaContext[],
```

Comment:
I wonder if schemaUsage is useful here ? as we are handling requestBody, it should be obvious to be SchemaContext.Input ?

Reply:
Good catch! I removed it.